packages/hw-app-ada/src/Ada.js

```diff
@@ -87,78 +86,47 @@ export default class Ada {
   }
 
   /**
-   * @description Get the root extended public key of the wallet,
-   * also known as the wallet recovery passphrase.
-   * BIP 32 Path M 44' /1815'
-   * 32 Byte Public Key
-   * 32 Byte Chain Code
+   * @description Get a public key from the specified BIP 32 path.
    *
-   * @return {Promise<{ publicKey:string, chainCode:string }>} The result object containing the root wallet public key and chaincode.
+   * @param {Array<number>} indexes The path indexes. Path must begin with `44'/1815'/n'`, and may be up to 10 indexes long.
+   * @return {Promise<{ publicKey:string, chainCode:string }>} The public key with chaincode for the given path.
    *
-   * @example
-   * const { publicKey, chainCode } = await ada.getWalletRecoveryPassphrase();
-   * console.log(publicKey);
-   * console.log(chainCode);
-   *
-   */
-  async getWalletRecoveryPassphrase(): Promise<{
-    publicKey: string,
-    chainCode: string
-  }> {
-    const response = await this.transport.send(
-      CLA,
-      INS_GET_PUBLIC_KEY,
-      0x01,
-      0x00
-    );
-
-    const [publicKeyLength] = response;
-    const publicKey = response.slice(1, 1 + publicKeyLength).toString("hex");
-    const chainCode = response
-      .slice(1 + publicKeyLength, 1 + publicKeyLength + 32)
-      .toString("hex");
-
-    return { publicKey, chainCode };
-  }
-
-  /**
-   * @description Get a public key from the specified BIP 32 index.
-   * The BIP 32 index is from the path at `44'/1815'/0'/[index]`.
-   *
-   * @param {number} index The index to retrieve.
-   * @return {Promise<{ publicKey:string }>} The public key for the given index.
-   *
-   * @throws 5201 - Non-hardened index passed in, Index < 0x80000000
-   * @throws 5202 - Invalid header
-   * @throws 5003 - Index not a number
+   * @throws 5003 - Some of the indexes is not a number
    *
    * @example
-   * const { publicKey } = await ada.getWalletPublicKeyWithIndex(0xC001CODE);
+   * const { publicKey, chainCode } = await ada.getExtendedPublicKey([ 0x80000000 + 44, 0x80000000 + 1815, 0x80000000 + 1 ]);
    * console.log(publicKey);
    *
    */
-  async getWalletPublicKeyWithIndex(
-    index: number
-  ): Promise<{ publicKey: string }> {
-    if (isNaN(index)) {
+  async getExtendedPublicKey(
+    indexes: Array<number>
+  ): Promise<{ publicKey: string, chainCode: string }> {
+    if (indexes.some(index => isNaN(index))) {
       throw new TransportStatusError(INDEX_NAN);
```

Comment:
Is this compatible with how BTC/other currencies check the validity of input? For that matter, do they expect Array<number> ?